web-crawler.js

```diff
@@ -234,6 +203,50 @@ function startRequestListeners() {
             mainFrameOriginTopHosts[tabID] = null;
         }
     });
+
+    // start our port listener
+    chrome.runtime.onMessage.addListener(function(request, sender, sendResponse) {
+        var profileCheck;
+
+        // if we've got a profiling command
+        if (typeof request.profiling !== 'undefined') {
+            // note the new profiling flag
+            profiling = request.profiling;
+        }
+
+         // if we're just checking to see if we're currenlty profiling
+        if(request.profileCheck) {
+            // send the value of profiling
+            sendResponse({ "isProfiling": profiling });
+        } else if (request.getOverview) { // if we're requested to return overview benchmarks
+            var overviewBenchmarks = {
+                fileSize: (totalFileSize+lastFileSize),
+                networkTime: (totalNetworkTime+lastNetworkTime),
+                assetCount: (totalAssetCount+lastAssetCount)
+            };
+
+            overviewBenchmarks = JSON.stringify(overviewBenchmarks);
+
+            sendResponse({ "overviewBenchmarks": overviewBenchmarks });
+        } else if (request.deleteOverview) {
+            totalAssetCount = 0;
+            totalFileSize = 0;
+            totalNetworkTime = 0;
+
+            lastAssetCount = 0;
+            lastFileSize = 0;
+            lastNetworkTime = 0;
+
+            sendResponse({ "deletedOverview": true})
+        } else if (!profiling) { // if we are supposed to stop profiling
+            var JSONString = stringifyAssetStore(profileStorage, false);
+            // send the profiling data
+            sendResponse({ "profiles": JSONString });
+
+            // and clear the profiling storage
+            profileStorage.clear();
+        }
+    });
```

Comment:
each `request.condition` must be in separate if-loops and each exiting from a `return; `